src/routes/home/__tests__/HomeScreen.spec.tsx

```diff
@@ -40,12 +40,15 @@ describe('HomeScreen', () => {
     mocked(useLoadDiscipline)
       .mockReturnValueOnce(getReturnOf(mockDisciplines()[0]))
       .mockReturnValueOnce(getReturnOf(mockDisciplines()[1]))
+      .mockReturnValueOnce(getReturnOf(mockDisciplines()[2]))
     mocked(useReadProgress).mockReturnValue(getReturnOf(0))
     const { findByText, getByText } = render(<HomeScreen navigation={navigation} />)
     const firstDiscipline = await findByText('First Discipline')
-    const secondDiscipline = getByText('Second Discipline')
+    const secondDiscipline = await findByText('Second Discipline')
```

Comment:
Can't you `getByText` this?

Reply:
I get a strange error/warning when using getByText

 console.error
    Warning: An update to ProfessionDetails inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
        at ProfessionDetails (/home/linuxdev/IdeaProjects/lunes-app/src/routes/home/components/ProfessionDetails.tsx:30:3)

Reply:
Ah you're right. But you'll only need to `findByText` the first discipline. For the others you can use `getByText`.

Reply:
As an explanation: In the first render the hook will return the discipline. Before the discipline array was empty. Now it is filled with disciplines. This change again triggers a rerender. If you try to do something in this time frame you'll need to wrap it in act. Generally render is wrapped in act, but the rerender happens after the first render. Basically when you try to get the first discipline in the test. This causes this warning/error. As a solution you can use findByText, which will correctly await the rerender and then return the found component. After the first findByText, the content should be completely available so you can the call getByText.

Hope that helps ^^

Reply:
And using findBy solves the warning and is  recommended here https://stackoverflow.com/questions/60115885/how-to-solve-the-update-was-not-wrapped-in-act-warning-in-testing-library-re.